### DISCUSSION TITLE
Installation of private repo via composer fails in Github codespaces

### DISCUSSION THREAD

**Discussion by loopy3025** [community/community#51851] [Codespaces]
### Private Repos in Github Codespaces

We are creating a codespace for one of our websites. All of our websites need to install code from one of our private repos via composer. According to the codespaces docs [https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces](https://docs.github.com/en/codespaces/managing-your-codespaces/managing-repository-access-for-your-codespaces), all we should need to do is add the private repo to the list in devcontainer.json...

```
        "codespaces": {
            "repositories": {
                "our-organization/private-repo": {
                    "permissions": "read-all"
                }
            }
        }
```

...then codespaces is supposed to pop up with some kind of access dialog after rebuild. When we rebuild, however, we get no access dialog. Then, when we run `composer install`, when the terminal tries to install the package from our private repo, it still complains and demands I create a personal access token. Everyone having to install a temporary personal access token is not a sustainable solution.

Are we required to install ssh keys to connect the private repo and this site's codespace? It'd be much easier if the devcontainer settings worked as described in the documentation. I am not sure what step I am missing. Could it be because I am doing it in my fork?

**Comment by samruddhikhandale** [comment]
Hi 👋 

Unfortunately, a `rebuild` won't pop the dialog and needs a new codespace creation. If you could git push the dev container changes and create a new codespace, then you should be able to fix the permission issue.

ℹ️ Also, the snippet pasted above needs to be wrapped within `customizations` property. 

I have opened an internal issue tracking this, also I can see if we could get the docs updated with this information, thanks!

**Comment by loopy3025** [comment]
Thanks so much for the response. Fortunately, we do have the file wrapped correctly. I committed my changes and pushed it to the repo. Upon creating a new codespace, we did receive this dialog:

<img width="651" alt="Screenshot 2023-04-03 at 9 22 02 PM" src="https://user-images.githubusercontent.com/6729935/229662087-f39738b5-1624-44c8-833c-428fe9ef0482.png">

When I click "Authorize", it just fires up the container. There is no additional access challenge for the other repo. I double checked and composer install still wanted a personal access token.

Is it because my ingesting repo is a fork? The repo that it's a clone of is the same organization as the component we're trying to add, but the fork is under my personal account.

Is there something I need to do on the component repo to make it available?

**Comment by hakre** [comment]
X-Ref: https://stackoverflow.com/q/75921119/367456
